tools/test-generator/src/main/java/org/citrusframework/generate/provider/http/ReceiveHttpResponseActionProvider.java

```diff
@@ -40,8 +40,8 @@ public ReceiveResponseModel getAction(String endpoint, HttpMessage message) {
         response.setBody(body);
 
         ReceiveResponseModel.Headers responseHeaders = new ReceiveResponseModel.Headers();
-        responseHeaders.setStatus(message.getStatusCode().toString());
-        responseHeaders.setReasonPhrase(message.getStatusCode().getReasonPhrase());
+        responseHeaders.setStatus(Integer.toString(message.getStatusCode().value()));
```

Comment:
message.getStatusCode().toString() returned a string. resolve the HttpStatusCode, check if it is HttpStatus and use the name and phrase if applicable. Otherwise chose default name "Status<YourCustomStatusCode>" and phrase "Custom status code <YourCustomStatusCode>"